STIPS/STIP-002.md

```diff
@@ -0,0 +1,150 @@
+# STIP-[002]: Fix aToken issuance bug
+*Using template v0.1*
+## Abstract
+Issuance modules including BasicIssuanceModule and DebtIssuanceModule revert while trying to issue SetTokens with Aave's aToken as components.
+## Motivation
+We need to fix this bug to launch our next set of structured products which would include SetTokens holding aTokens as components. Fixing this bug would also enable third party managers to gain exposure to yield bearing Aave aTokens in their SetTokens.
+## Background Information
+In all our issuance modules, during issuing a SetToken, we make sure the sum of existing balance of the component token in the SetToken and the quantity of component token being transferred from the issuer to the SetToken is equal to the new balance of the component token in the SetToken.
+```javascript
+uint256 existingBalance = _componentToken.balanceOf(_setToken);
+SafeERC20.transferFrom(
+    _componentToken,
+    _issuer,    // from
+    _setToken,  // to
+    _quantity
+);
+uint256 newBalance = _componentToken.balanceOf(_setToken);
+require(newBalance == existingBalance.add(_quantity), "Invalid post transfer balance");
+```
+Similarly, during redemption of a SetToken, we make sure the quantity of component token being transferred from the SetToken to the redeemer subtracted from the old component token balance of the SetToken is equal to new component token balance of the SetToken.
+In both cases, if the requirement is not met then we revert with "Invalid post transfer balance" message.
+
+- In _DebtIssuanceModule#issue_ and _BasicIssuanceModule#issue_, we transfer the component token from the issuer to the SetToken using the _ModuleBase#transferFrom_ function, which internally calls the _ExplicitERC20#transferFrom_ function, which performs the above mentioned check.
+- In _DebtIssuanceModule#redeem_ and _BasicIssuanceModule#redeem_, we transfer the component from the SetToken to the redeemer using _Invoke#strictInvokeTransfer_ library function which internally calls _ExplicitERC20#transferFrom_ function, which performs the above mentioned check.
```

Comment:
Let's be careful here, `strictInvokeTransfer` doesn't call `ExplicitERC20.transferFrom`, so we need to apply a fix in two places it seems. One on `transfer` and another on `transferFrom`.